fluentd-forward/README.md

```diff
@@ -0,0 +1,355 @@
+
+### <a name="user"></a>Create user and role for access audit log events
+
+Log into Teleport Authentication Server, this is where you normally run tctl. Create a new user and role that only has read-only API access to the `event` API. The below script will create a [yaml resource file](example/fluentd-forward.yaml) for a new user and role.
+
+```yaml
+kind: user
+metadata:
+  name: fluentd-forward
+spec:
+  roles: ['fluentd-forward']
+version: v2
+---
+kind: role
+metadata:
+  name: fluentd-forward
+spec:
+  allow:
+    rules:
+      - resources: ['event']
+        verbs: ['list','read']
+version: v3
+```
+
+Here and below follow along and create yaml resources using `tctl create -f`:
+
+```sh
+tctl create -f fluentd-forward.yaml
+```
+
+### Export fluentd-forward Certificate
+
+Teleport Plugin use the fluentd role and user to read the events. We export the identity files, using tctl auth sign.
+
+```sh
+tctl auth sign --format=tls --user=fluentd-forward --out=fd --ttl=720h
+```
+
+This will generate `fd.cas`, `fd.crt` and `fd.key` which will be used to connect plugin to your Teleport instance.
+
+## Installing the plugin
+
+We recommend installing the Teleport Plugins alongside the Teleport Proxy. This is an ideal location as plugins have a low memory footprint, and will require both public internet access and Teleport Auth access. We currently only provide linux-amd64 binaries, you can also compile these plugins from source.
+
+### Install the plugin from source
+
+Please ensure that Docker is running!
+
+```sh
+git clone https://github.com/gravitational/teleport-plugins.git --depth 1
+cd teleport-plugins/fluentd-forward/build.assets
+make install
+```
+
+This command will build `build/fluentd-forward` executable and place it to `/usr/local/bin` folder. The following error means that you do not have write permissions on target folder:
+
+```sh
+cp: /usr/local/bin/fluentd-forward: Operation not permitted
+```
+
+To fix this, you can either set target folder to something listed in your `$PATH`:
+
+```sh
+make install BINDIR=/tmp/test-fluentd-setup
+```
+
+or copy binary file manually with `sudo`:
+
+```sh
+sudo cp build/fluentd-forward /usr/local/bin
+```
+
+### Configure the plugin
+
+Save the following content to `fluentd-forward.toml`:
+
+```toml
+storage = "./fluentd-forward-storage" # Plugin will save it's state here
+timeout = "10s"
+batch = 10
+namespace = "default"
+
+[fluentd]
+cert = "client.crt"
+key = "client.key" 
+ca = "ca.crt"
+url = "https://localhost:8888/test.log"
+
+[teleport]
+addr = "localhost:3025"
+ca = "fd.cas" 
+cert = "fd.crt"
+key  = "fd.key"
+```
+
+## Setup fluentd
+
+For the purpose of testing, we will run the local fluentd instance first.
+
+### <a name="mtls"></a>Generate mTLS certificates
+
+For the purpose of security, we require mTLS to be enabled on the fluentd side. You are going to need [OpenSSL configuration file](example/ssl.conf). Put the following contents to `ssl.conf`:
```

Comment:
This is going to be a slow down for getting started, can you please create a fluentd-forward configure command that works similar to:

https://goteleport.com/docs/application-access/getting-started/#step-23-install-and-configure-teleport

and generates all certificates and files for easy to get started? Let's aim for one liner that sets everything up for a local config. We can have advanced section that will show how to generate certificates, etc

Reply:
OK. Actually, there is one: `make gen-example-mtls`. It might be enough when plugin is built from source, but might not for binary version. I'll try to add the equivalent to binary itself (or possibly to separate bash script if that's okay).

Reply:
Let's keep everything in one binary - that's what people will be downloading, you can generate certs using go much more conveniently than using external tools, see for example teleport code:

https://github.com/gravitational/teleport/blob/8922af0c6191af8fdf251acfb80f5e8789b07cd0/lib/utils/tls.go#L93

Reply:
This is done. mTLS certs are not self-signed, they require CA and separate client/server certs to be generated as well.